### PR TITLE
chore: split tests, Javadoc and Spotless

### PR DESCRIPTION
**Pull request by JoergSiebahn**

### CODE REVIEW

File: .github/workflows/javadoc.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This workflow builds the Javadoc
+name: Javadoc
```
Comment (christopher-cudennec) [inline .github/workflows/javadoc.yml:2]: Don't we need to upload the generated artifacts? 
I guess they should also be published on Maven Central.

Comment (JoergSiebahn) [reply .github/workflows/javadoc.yml:2]: We store these artifacts in Java CI:

```
            build/reports/
            */build/test-results/**/*.xml
```

None of these contain the generated Javadoc.
Is there actually anything that we use from the build for publication?

Comment (JoergSiebahn) [reply .github/workflows/javadoc.yml:2]: At least the snapshot repo has Javadoc jars with content.

image

Comment (christopher-cudennec) [reply .github/workflows/javadoc.yml:2]: You're right. We seem to build all artifacts for publication in that specific workflow.

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**